src/Unistep2.cpp

```diff
@@ -37,12 +35,13 @@ Unistep2::Unistep2(int _p1,int _p2, int _p3, int _p4, int _steps, unsigned long
 // function. Or else return quickly so that we don't stall the loop.
 boolean Unistep2::run()
 {
-  if (!stepstogo) // will be true if zero (!false = true)
-  {//we're done
-    return true;
-  } else {
+  if (stepstogo)
```

Comment:
There's a reason this block was written the way it was. I know it's not stylistically the best, but it's effective and makes that evaluation the quickest when it's false.